### DISCUSSION TITLE
Inconsistent results for repo API queries

### DISCUSSION THREAD

**Discussion by jbreckmckye** [community/community#47552] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello all,

I'm having trouble fetching a list of my organization's repositories with both the REST and GraphQL APIs. The issue is with both, so I wonder if the problem is further up the chain.

In a nutshell: certain repos simply aren't returned by the API calls, despite paginating through the entire set; some are only returned sometimes, inconsistently. This seems the case despite using different PATs.

Here's an example call using the REST API via SDK:
```typescript
const iterator = octokit.paginate.iterator(octokit.rest.repos.listForOrg, {
      type: 'private',
      direction: 'asc',
      sort: 'full_name',
      org: ORGANIZATION_NAME,
      per_page: 100,
    })
```

This returns 733 responses for my org today - but in the UI, 833...

![image](https://user-images.githubusercontent.com/3148617/219039692-3f1d4227-4de2-43cd-841c-d1873e5b792e.png)

And not just that - several of the items returned by the iterator are incorrect. There's a whole bunch of what appear to be forks of our code by non-org users, which isn't what I understand `private` to mean here.

My first assumption was that they key I was using might not have enough scope - but even using my own issued PAT, the query is missing org repositories that I created and am an admin of.

The project I'm working on needs a reliable list of repositories, in order to mirror state and capture important metrics about code at our organisation. What can I do to get this consistent list of organisation repositories?

---

EDIT

One thing I discovered is that if I stopped using my existing PAT, and instead created a brand new personal access token with exactly the same scopes, now the query seems to retrieve a larger (possibly full?) set of results.

Is it possible there is something odd happening with caches?

**Comment by unknowVariable** [comment]
It sounds like you are experiencing an issue where your API queries to fetch the list of repositories for your organization are not returning all of the expected results, and in some cases are returning incorrect results. This can be frustrating, especially if you are relying on this information for important business needs.

Here are a few things you can try to resolve the issue:

1. Check the authentication and authorization: Ensure that the personal access token (PAT) you are using to authenticate the API requests has sufficient scopes to read the repositories for the organization. You can check the scopes of the PAT by going to your account settings on GitHub and viewing the token details.

2. Check the filters and parameters: Ensure that you are using the correct filters and parameters in your API requests. For example, if you are filtering by a specific type of repository (e.g., private), this might affect the number of repositories returned.

3. Check the pagination: Make sure that you are paginating through all the pages of results to ensure you are not missing any repositories. You can use the page and per_page parameters to specify the page number and the number of results per page.

4. Check for rate limiting: GitHub API has rate limits which may affect your ability to retrieve all of the repositories in a single API call. If you exceed the rate limit, you may need to wait before retrying your API call.

5. Check for known issues: It's possible that there are known issues or bugs with the GitHub API that are affecting your API queries. You can check the GitHub API status page for any known issues or outages.

6. Contact GitHub support: If you have tried all of the above steps and are still experiencing issues with the API queries, you may want to contact GitHub support for further assistance. They may be able to help you troubleshoot the issue and provide a solution.

I hope this helps you resolve the issue and retrieve a consistent list of organization repositories.

**Reply by jbreckmckye** [reply]
Hi @alixconstantin, thanks for getting back so quickly.

I did consider some of these options, including the possibility of the token being problematic.

One thing I discovered is that if I stopped using my existing PAT, and instead created a _brand new personal access token with exactly the same scopes_, now the query seems to retrieve a larger (possibly full?) set of results.

Is it possible there is something odd happening with caches?

**Comment by Raicuparta** [comment]
Hey @jbreckmckye, did you ever come to any conclusion here? I think I might have a similar problem fetching releases.

Also, that other reply you got is ChatGPT so uh, yeah.

**Reply by DropSnorz** [reply]
Hello @Raicuparta . I don't know if it's related to your problem, but I noticed I have some "ghost" releases in one of my repositories. 

https://github.com/orgs/community/discussions/76962

**Reply by jbreckmckye** [reply]
No, I ended up just renewing the token. For a while after if I used the old token repos would be missing, so I figured something must be being cached somewhere

**Comment by uladzimirbabrou-tomtom** [comment]
I'm using Search API (by private GitHub App) to list repos of our organisation and getting different results all the time.
Even if I use search box on the organisation home page to find repos, I get 177, 169, 166, 157, 154, 150, ... results for the same input string almost every time.

It makes impossible to rely on GitHub Search API even for Enterprise users.

**Comment by asocorrocognex** [comment]
Anyone noticing a similar kind of inconsistency with the CoPilot metrics API?  We are getting different values for the same dates.
